sw/device/lib/crypto/impl/ecc/p256.h

```diff
@@ -230,6 +230,27 @@ status_t p256_sideload_keygen_start(void);
 OT_WARN_UNUSED_RESULT
 status_t p256_sideload_keygen_finalize(p256_point_t *public_key);
 
+/**
+ * Start an async ECDSA/P-256 signature generation operation on OTBN.
+ *
+ * This function allows for configuration of the secret scalar k.
+ * KATs for FIPS CMVP compliance require ECDSA implementations to receive
+ * non-random selected ephemeral k as input (see p.82 bottom in Implementation
+ * Guidance for FIPS 140-3 and the Cryptographic Module Validation Program).
+ * https://csrc.nist.gov/csrc/media/Projects/cryptographic-module-validation-program/documents/fips%20140-3/FIPS%20140-3%20IG.pdf
+ *
+ * Returns an `OTCRYPTO_ASYNC_INCOMPLETE` error if OTBN is busy.
+ *
+ * @param digest Digest of the message to sign.
+ * @param private_key Secret key to sign the message with.
+ * @param private_key Secret scalar k.
```

Comment:
@h-filali can we please indicate what sizes of secret scalar k we tolerate here?